Further clarifications on HCS vs FHIR security label

mojitoj opened the pull request:
there are a number of tickets around this which I am hoping to address with one broad PR.

Will resolve: 
- https://jira.hl7.org/browse/FHIR-33522
- https://jira.hl7.org/browse/FHIR-33514
- https://jira.hl7.org/browse/FHIR-28033
- https://jira.hl7.org/browse/FHIR-27093

mojitoj committed:
overall edits and terminology caveats section

JohnMoehrke commented on input/pagecontent/index.md:
````diff
@@ -1,15 +1,86 @@
+```
+
+The HCS, however, defines Security Label as a different, more complex structure based on the concept of [Named Tag Set](glossary.html#named-tag-set). As implied by the title, a _Named Tag Set_ is a collection of _Tags_ together with a _Tag Set Name_. Each _Tag_ is a simple code and the _Tag Set Name_ assigns a name to the collection. Conceptually, this is equivalent to a JSON object similar to th ...
+```json
````
Could we just call this abstract, rather than json? It would otherwise look the same, but would just be abstract.  I wonder if asking Grahame might help.

JohnMoehrke commented on input/pagecontent/index.md:
````diff
@@ -1,15 +1,86 @@
+```
+As one can immediately notice, `TagSetName` is similar (although not quite conceptually identical) to the `system` attribute in a `Coding` data structure for a FHIR security label. Part of the contribution of this IG is to provide guidance on how to map the abstract structures defined by the HCS to FHIR. This is further discussed in detail in the section on [Security Labeling Conceptual Structure ...
+
+Another terminology caveat is the difference between _Tags_ in HCS and in FHIR. In FHIR, `meta.tag` is a data structure for capturing general metadata and observations about a resource and is a collection of objects of type [`Coding`](https://build.fhir.org/datatypes.html#Coding). _Tags_ in HCS, however, are opaque values (equivalent to the `code` in a `Coding` data structure).
````
not clear why you are bringing in meta.tag.  I can guess that you think there might be confusion, but I really don't think our reader is going to be confused.

mojitoj replied:
yes, my intention was to clear up any confusion about these terms as they're used in FHIR vs. in HCS.

JohnMoehrke commented on input/pagecontent/index.md:
```diff
@@ -1,15 +1,86 @@
+
+- Granular data segmentation at the sub-resource level, known as [inline labeling](inline.html).
+
+For example, defining the [`sec-label-basis` extension](StructureDefinition-extension-sec-label-basis.html) in this IG enables an implementer to group relevant security labels in `Resource.meta.security` into an HCS-conformant Security Labels, with  each applicable policy represented as a separate named tag set. For example, when the information in a resource is governed by both US _42 CFR Part 2 ...
```
bringing in just one example is confusing. I think there should be no example here. The bullets are clear enough at this level of the document.

mojitoj replied:
I agree. Seems like this very example is also given in `spec.md`

JohnMoehrke commented on input/pagecontent/index.md:
```diff
@@ -1,15 +1,86 @@
+From the outset, FHIR has supported security labels with a dedicated optional element ([`Resource.meta.security`](https://build.fhir.org/resource-definitions.html#Meta.security)) that can appear on any resource (regardless of resource type), therefore, enabling recording and processing of security labels in a unified manner. 
+### Terminology Caveats
+In FHIR, a security label is defined as a simple value of type [`Coding`](https://build.fhir.org/datatypes.html#Coding) which is composed of (among other details), a _code_ and the identifier of the _system_ in which the code is defined, together with a human readable _display_ name for the _code_. For example, the following data structure represents a FHIR security label for _restricted_ confide ...
+
```
use of italics for .code, .system, and .display -- You should have these indicted as code with back quote.

JohnMoehrke reviewed (commented):
I put in some comments. Let me know if you need clarifiction.

mojitoj committed:
@John Moehrke 's comments

mojitoj committed:
restructure concept.md, add new diagrams, and move detailed discussions from index.md to concept.md.

mojitoj commented:
Thanks @JohnMoehrke. I incorporated all your comments --except the one about `tag`. I think it's worth mentioning the difference and at least it won't hurt since we have direct hyper links to `meta` data structure which also includes `tag`. 

I also moved the details to the `concept` page and added a couple of diagrams. I'm wondering if I got your comment about `abstract` data type right so I appreciate if you could let me know if you meant something else --I'm also thinking whether the abstract data structure is needed considering the diagrams might provide enough clarification.

mojitoj committed:
edit: whitespace cleanup
